broken-site/broken-site-store/src/main/java/com/duckduckgo/brokensite/store/BrokenSiteDao.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2023 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.brokensite.store
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.OnConflictStrategy
+import androidx.room.Query
+import androidx.room.Transaction
+import androidx.room.Update
+import com.duckduckgo.app.global.formatters.time.DatabaseDateFormatter
+
+@Dao
+abstract class BrokenSiteDao {
+
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
```

Comment:
I think the DAO can be simplified.

I'd make `hostnameHashPrefix` the primary key because it has to be unique, this makes the `update()` fun not needed since insert will replace when finding a duplicate which also means we can get rid of `upsertBrokenSiteReport()`.

The other transactions can also be deleted and simplified with one query. So I would delete the `update()`, `getAllExpiredIds()`, `upsertBrokenSiteReport()` and `cleanupBrokenSiteReport()` functions and instead do something like this

```
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract fun insertBrokenSiteReport(brokenSiteReportEntity: BrokenSiteLastSentReportEntity)

    @Query("select * from broken_site_last_sent_report where hostnameHashPrefix = :hostnameHashPrefix")
    abstract fun getBrokenSiteReport(hostnameHashPrefix: String): BrokenSiteLastSentReportEntity?

    @Query("delete from broken_site_last_sent_report where lastSentTimestamp < :expiryTime")
    abstract fun deleteAllExpiredIds(expiryTime: String)
```